cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -125,6 +130,146 @@ _CCCL_DEVICE void transform_kernel_impl(
   }
 }
 
+template <int Length>
+_CCCL_HOST_DEVICE _CCCL_CONSTEVAL auto load_store_type()
+{
+  if constexpr (Length == 1)
+  {
+    return ::cuda::std::int8_t{};
+  }
+  else if constexpr (Length == 2)
+  {
+    return ::cuda::std::int16_t{};
+  }
+  else if constexpr (Length == 4)
+  {
+    return ::cuda::std::int32_t{};
+  }
+  else if constexpr (Length == 8)
+  {
+    return ::cuda::std::int64_t{};
+  }
+  else if constexpr (Length == 16)
+  {
+    return int4{};
+  }
```

Comment:
Should this use `int128` if available?

Reply:
I don't know. @gevtushenko? What's important is that a 128-bit load is generated.